### PR TITLE
relax matching rules for gems to support versioned dependencies

### PR DESCRIPTION
**Pull request by huerlisi**
In the Gemfile one may specify a version (or other arguments) when
declaring dependencies. Those dependencies were not detected with the
rule set.

This patch relaxes the RegEx used to detect gem dependencies by dropping
the `$`.

NOTE: only the test for one framework (Rails) has been adapted.

### CODE REVIEW

File: spec/frameworks_rules_spec.rb [hunk 1]
```diff
@@ -14,6 +14,12 @@
       expect(repository.secondary_frameworks).to eq []
     end
 
+    it 'returns Rails when specified with version' do
+      repository = repository('Gemfile', "bogus\ngem 'rails', '~> 4.3'\nbogus")
```
Comment (koffeinfrei) [inline spec/frameworks_rules_spec.rb]: Could you add that to all the examples that cover the Gemfile rule?

**Comment by koffeinfrei** [comment]
You're totally right. Thanks!

**Comment by huerlisi** [comment]
Yes i can add that.

But I'm actually working on support for gems in `group xxx do` blocks now. That includes a spec that it can have leading whitespace, but also that it is not detected when commented out. This will lead to two additional specs. This means then have 4 specs per gem detection that should be duplicated by the number of rules. That's a lot of code duplication.

I actually think about implementing a helper for gem detection. Then we only would need to write those specs once...

**Comment by koffeinfrei** [comment]
I thought about group support too, but as I see it this isn't a use case for the primary gems as rails and sinatra etc. They are always outside of a group. Do you see a use case where this is necessary?

**Comment by huerlisi** [comment]
While I was thinking about groups first it's general indentation support as per Ruby/Gemfile syntax actually.

I've added support for those cases and extended the specs.

**Comment by huerlisi** [comment]
Ready for review. Additional work will be done in other branches...

File: spec/frameworks_rules_spec.rb [hunk 2]
```diff
@@ -32,17 +53,66 @@
...
+      repository = repository('Gemfile', "bogus\ngem 'dashing', '~> 4.3'\nbogus")
+      expect(repository.primary_frameworks).to eq ['Sinatra']
+    end
+
+    it 'returns Sinatra when indented' do
+      repository = repository('Gemfile', "bogus\n  gem 'dashing'\nbogus")
+      expect(repository.primary_frameworks).to eq ['Sinatra']
+    end
+
+    it 'does not returns Sinatra when commented out' do
```
Comment (koffeinfrei) [inline spec/frameworks_rules_spec.rb]: -> "does not return"

Comment (huerlisi) [inline spec/frameworks_rules_spec.rb]: Fixed

**Comment by huerlisi** [comment]
Fixed the typo, can we merge this like it is and do a refactor of the rules later on?

**Comment by huerlisi** [comment]
Opened issue #4 regarding DRY rules.

**Pull request merged**